src/cascade/model/session.py

```diff
@@ -0,0 +1,552 @@
+        # From covariate name to the x_<number> name that is used internally.
+        # The session knows this piece of information but not the covariate
+        # reference values. This is here because the columns of avgint and data
+        # need to be renamed before writing, and they aren't part of the model.
+        self._covariate_rename = dict()
+        for create_name in ["data", "avgint"]:
+            setattr(self.dismod_file, create_name, self.dismod_file.empty_table(create_name))
+
+    def fit(self, model, data, initial_guess=None):
+        """This is a fit without a predict. If the model
+        has random effects, this optimizes over both fixed
+        and random effects.
+
+        Args:
+            model (Model): A model, possibly without scale vars.
+            data (pd.DataFrame): Data to fit.
+            initial_guess (Var): Starting point to look for solutions. If not
+                given, then the mean of the priors is taken as the initial
+                guess.
+
+        Returns:
+            DismodGroups[Var]: A set of fit var.
+        """
+        if model.random_effect:
+            MATHLOG.info(f"Running fit both.")
+            return self._fit("both", model, data, initial_guess)
+        else:
+            MATHLOG.info(f"Running fit fixed.")
+            return self._fit("fixed", model, data, initial_guess)
+
+    def fit_fixed(self, model, data, initial_guess=None):
+        """Fits a model without optimizing over any random effects.
+        It does apply constant child value priors, but other random effects
+        are constrained to zero. (This is equivalent to fitting with
+        ``bound_random`` equal to zero.) This is useful when one uses fitting
+        with no random effects as a starting point for fitting with
+        random effects.
+
+        Args:
+            model (Model): A model, possibly without scale vars.
+            data (pd.DataFrame): Data to fit.
+            initial_guess (Var): Starting point to look for solutions. If not
+                given, then the mean of the priors is taken as the initial
+                guess.
+
+        Returns:
+            DismodGroups[Var]: A set of fit var.
+        """
+        return self._fit("fixed", model, data, initial_guess)
+
+    def fit_random(self, model, data, initial_guess=None):
+        """
+        Fits the data with the model.
+        This optimizes the random effects with the fixed effects set to their
+        starting values. The fixed effects are unchanged.
+
+        Args:
+            model (Model): A model, possibly without scale vars.
+            data (pd.DataFrame): Data to fit.
+            initial_guess (Var): Starting point to look for solutions. If not
+                given, then the mean of the priors is taken as the initial
+                guess.
+
+        Returns:
+            DismodGroups[Var]: A set of fit var.
+        """
+        return self._fit("random", model, data, initial_guess)
```

Comment:
Should this check that there are random effects in the model? Or do we just wait for dismod to compalin?